fix: added feed_options as a keyword argument to GCSFeedStorage.

protokoul opened the pull request:
This PR attempts to resolve #6617

- Added feed_options as a keyword argument to `GCSFeedStorage` in its `__init__()` and `from_crawler()`.
- Added test cases to `GCSFeedStorageTest`
~~- Refactored class `GCSFeedStorage` to import and create gcs client in its `__init__` method~~
~~- Added `requires_gcs` to the markers list in pyproject.toml~~
~~- Created a fixture to check google-cloud-storage library in GCSFeedStorageTest tests just like checking boto3 in S3FeedStorageTest~~

Gallaecio commented on scrapy/extensions/feedexport.py:
```diff
@@ -276,20 +276,41 @@ def _store_in_thread(self, file: IO[bytes]) -> None:
 
+        if feed_options and feed_options.get("overwrite", True) is False:
+            logger.warning(
+                "S3 does not support appending to files. To "
```
This is not S3 :slightly_smiling_face:

protokoul replied:
No problem. I noticed that too just now.

protokoul commented:
Note to self: As part of the [initial commit](https://github.com/scrapy/scrapy/pull/6628/commits/9dd99879f1d016ba1e32c9cc1cf99d9bc81b6a9c) I initially tried to refactor the class `GCSFeedStorage` to import and create gcs client in its `__init__` method, added `requires_gcs` to the markers list in pyproject.toml, and created a fixture to check google-cloud-storage library in GCSFeedStorageTest tests just like checking boto3 in S3FeedStorageTest, but this failed two jobs that can be seen [here](https://github.com/scrapy/scrapy/actions/runs/12915253892/job/36016694886?pr=6628) and [here](https://github.com/scrapy/scrapy/actions/runs/12915253892/job/36016695863?pr=6628). In my next [commit](https://github.com/scrapy/scrapy/pull/6628/commits/596444df7a9603485b57256fe3114d02ceb18962), I tried to undo some changes but those two jobs still failed. After that, when I removed the gcs_client from `__init__()` and moved it to back to `_store_in_thread()`, then those two jobs passed as seen [here](https://github.com/scrapy/scrapy/pull/6628/commits/85c3b6a14909b1afdd2bec23e53488af961bd865).

wRAR commented:
Thanks!
